site/app/controllers/HomePageController.php

```diff
@@ -40,6 +42,79 @@ public function showLibrary() {
 		);
     }
 
+    /**
+     * @Route("/homework/library/search", methods={"GET"})
+     * @return Response
+     */
+    public function searchLibrary($path = null) {
+        if ($path === null) {
+            $path = $this->homework_path;
+        }
+        $gradeable_ids = FileUtils::getDirWithText($path, "config.json");
+        return Response::JsonOnlyResponse(
+            JsonResponse::getSuccessResponse($gradeable_ids)
+        );
+	}
```

Comment:
All style fixes will come along after merging submitty/master since they have updated coding standards as well as a coding sniffer tool that fixes style errors.